### ISSUE TITLE
Generic Connector Communication Error:

### ISSUE THREAD

**Issue by Subramanian-Palanisami** [Bilal-S/iis2tomcat#71]
Hi Bilal,
Good Day!!

Have found your **BonCode IIStoTomcat connecter** will work out for my requirement. Getting following error and it would be great if this issue resolved and thankful to you. 

**Generic Connector Communication Error:**
Please check and adjust your setup:
Ensure that Tomcat is running on given host and port.
If this is a timeout error consider adjusting IIS timeout by changing executionTimeout attribute in web.config (see manual).

Configuration Details:

1. IIS:
Under my main site, created virtual directory called(ea-test). 
Physical Path: C:\Tomcat\apache-tomcat-9.0.7\webapps\ea-test
Virtual-Path: /ea-test

So, my complete url is: **www.xyz.in/ea-test**

2. Tomcat Application deployed here:
C:\Tomcat\apache-tomcat-9.0.7\webapps\ea-test\index.html

3. BonCodeAJP13.settings
<Settings>
  <Port>8009</Port>
  <Server>localhost</Server>   
  <MaxConnections>0</MaxConnections>
  <LogLevel>0</LogLevel>  
  <FlushThreshold>0</FlushThreshold>
  <EnableRemoteAdmin>False</EnableRemoteAdmin>
</Settings>

Approaches tried to fix but not worked out:
1. Changing MaxConnections value to 0 and restart IIS.
2. Make sure connector port value is 8009

    <Connector port="8009" protocol="AJP/1.3" redirectPort="8443" />

Eagerly looking for your response.

Regards,
Subramanian Palanisami

**Comment by Bilal-S** [comment]
Subramanian,
on the surface your setup looks correct. Are you able to access your tomcat site directly via it's html port (8080 default). 

Also can you attempt to use `127.0.0.1` as server setting instead of `localhost`. 

Questions;
1) What version of connector are you using?
2) What is your logfile entry during the error (set `LogLevel` to 2) and set `LogDir` if needed.

Best,
Bilal

**Comment by Subramanian-Palanisami** [comment]
Hi Bilal,

YES, able to access  localhost:8080/ea-test as expected.

tried to use 127.0.0.1 as server setting instead of localhost, but no luck!

> <Settings>
>   <Port>8009</Port>
>   <Server>127.0.0.1</Server>   
>   <MaxConnections>0</MaxConnections>
>   <LogLevel>0</LogLevel>  
>   <FlushThreshold>0</FlushThreshold>
>   <EnableRemoteAdmin>False</EnableRemoteAdmin>
> </Settings>

Answers are here:
1. Get to know version is 1.0.28 from Release History file.
2. There wont be any specific error logging in Tomcat log. Kindly let me know how to set LogLevel to 2 .

Regards,
Subramanian Palanisami

**Comment by Subramanian-Palanisami** [comment]
Hi Bilal,

Good news!! Issue is fixed. steps have followed:

1. uninstall BonCode.
2. and install again fresh BondCode exe. this time i unchecked Servlet option from select Handler Mapping page. 
3. Restart website from IIS.

Thats it; Thanks you so much Bilal for your wonderful BonCode plugin. It is great stuff from get rid out of from url rewrite or ISAPI. Please close this issue.

**Have one more question for you. please.**
1. is it possible can i create one more virtual directory called (/eacc) under my main site (www.xyz.in) which need to connect tomcat application : C:\Tomcat\apache-tomcat-9.0.7\webapps\eacc. 

Basically eacc is a spring boot data api. it does not have any html or cfm pages. just using this for connect to db and return json as response.

Please let me know will this approach will work out? do i need to do any special configuration for this?
